backend_ts/src/stringParsers.ts

```
@@ -0,0 +1,243 @@
+      // Check distant surroundings
+      const distantSurroundings = this.envState.surroundings.getDistantSurroundings();
+      if (!distantSurroundings) return null;
+
+      const directionData = distantSurroundings.get(direction as unknown as Direction);
+      if (!directionData) continue;
+
+      switch (thingType) {
+        case "block": {
+          const found = directionData.blocksToClosestCoords?.get(thing)
+          if (found != null) return found;
+        }
```

Biome (1.9.4)

[error] This case is falling through to the next case.

Add a `break` or `return` statement to the end of this case to prevent fallthrough.

(lint/suspicious/noFallthroughSwitchClause)